File: Math/fatorialRecursive/fatorial_recursive.java
public class fatorial_recursive {

    public static void main(String[] args) {
		int Fat;
		Fat = fat(5);
		System.out.println(Fat);
    }
    public static int fat(int n){
    	if (n == 0 || n == 1){
    		return 1;
		}
    	else{
    		return n*fat(n-1);
		}
	}
}

File: Math/fatorialRecursive/fatorial_recursive.cpp
#include <iostream>
using namespace std;

int fatorialInstance(int fatorialNumber){
    if (fatorialNumber == 0 || fatorialNumber == 1){
        return 1;
    }
    else{
        return fatorialInstance*fatorialExpression(fatorialInstance-1);
    }
}
int main(){
    int inputNumber;
    cout << "Insert a number to factor: " << endl;
    cin >> inputNumber;
    cout<< "The factorial of: "<< inputNumber << " is " <<fat(n);
}
